generator/parseXML/ParseXML.py

```diff
@@ -247,8 +262,17 @@ def get_typecode(self, node):
         return typecode
 
     def analyse_enumname(self, enum_name):
+        """
+        Pulls out "parts" of the name of an <enum> node, depending on
+        whether an upper-case letter is found in the name or not.
+
+        TODO: why?
```

Comment:
SK needs to look at this and came back and answer

Reply:
So this one is due to the different styles of naming in an enum. We want the code to end up with a name that is
FOO_BAR_XYZ but we cannot guarantee that these have been entered this way. I used cname (classname) and tname (typename) to distinguish.

So you might have BOUNDARYKIND or BOUNDARY_KIND or BoundaryKind; which are all the classname and so we want

typedef enum
{
  SPATIAL_BOUNDARYKIND_ROBIN_VALUE_COEFFICIENT                        /*!< The spatial boundarykind is @c "Robin_valueCoefficient". */
, SPATIAL_BOUNDARYKIND_ROBIN_INWARD_NORMAL_GRADIENT_COEFFICIENT       /*!< The spatial boundarykind is @c "Robin_inwardNormalGradientCoefficient". */
, SPATIAL_BOUNDARYKIND_ROBIN_SUM                                      /*!< The spatial boundarykind is @c "Robin_sum". */
, SPATIAL_BOUNDARYKIND_NEUMANN                                        /*!< The spatial boundarykind is @c "Neumann". */
, SPATIAL_BOUNDARYKIND_DIRICHLET                                      /*!< The spatial boundarykind is @c "Dirichlet". */
, SPATIAL_BOUNDARYKIND_INVALID                                        /*!< Invalid BoundaryKind value. */
} BoundaryKind_t;

so first two parts are always the same and then the part after second _ is value - which might have underscores.

Hope this makes sense.

Reply:
Thanks Sarah. I will add this comment into the doc for that function